File: 594 - Longest Harmonious Subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findLHS = function(nums) {
    let map = new Map();
    for (let num of nums) {
        if (map.has(num)) {
            map.set(num, map.get(num) + 1);
        } else {
            map.set(num, 1);
        }
    }
    let res = 0;
    for (let [key, value] of map) {
        if (map.has(key + 1)) {
            res = Math.max(res, map.get(key + 1) + value);
        }
    }
    return res;
};

File: 594 - Longest Harmonious Subsequence.java
import java.util.HashMap;

class Solution {
    public int findLHS(int[] nums) {
        HashMap<Integer, Integer> m = new HashMap<>();
        int res = 0;
        for (int num: nums){
            m.put(num, m.getOrDefault(num, 0) + 1);
        }
        Integer cj;
        for (int num: m.keySet()) {
            if ((cj = m.get(num + 1)) != null) {
                res = Math.max(res, m.get(num) + cj);
            }
        }
        return res;
    }
}